### PR TITLE
Test package scaffolding on CI

### PR DESCRIPTION
**Pull request by Ilia-Kosenkov**
`use_extendr()` prepares new packages to work with extendr. It is useful to test if `use_extendr()` creates all the necessary files.
To do that, we can create an empty project, link it to extendr, and write a simple test. If R CMD check on this package succeeds, the setup was correct.

See #54 for a discussion on how to correctly build a package with `{rextendr}`.

### CODE REVIEW

**Comment by clauswilke** [comment]
Somehow it's shown all sorts of changes that were already committed on the main branch. Did you do a merge commit with main? If not, I suggest you do one. That's always the easiest way to keep branches and main synchronized, and it'll disappear when we do the squash merge.

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -102,3 +106,36 @@ jobs:
...
+              c(
+                  "test_that(\"`hello_world()` works\", {",
+                  "  expect_equal(hello_world(), \"Hello world!\")",
+                  "})"
+              ),
+              file.path("tests", "testthat", "test-hello.R")
+          )
+          rextendr::use_extendr()
+          devtools::document()
+          devtools::install()
```
Comment (clauswilke) [inline .github/workflows/R-CMD-check.yaml:132]: I think you can use `devtools::install(quick = TRUE)` here.

**Comment by Ilia-Kosenkov** [comment]
@clauswilke,
Oh sorry, I was in a hurry and did it wrong, I will redo it shortly.

File: .github/workflows/R-CMD-check.yaml (hunk 1 above)
Comment (Ilia-Kosenkov) [reply .github/workflows/R-CMD-check.yaml]: I fixed this.

**Comment by clauswilke** [comment]
The latest version looks fine to me, but as is it doesn't test that `register_extendr()` actually updates the wrappers.

One way to do this, I think, is to test that the first line in the wrappers is `# Generated by extendr: Do not edit by hand`. This line gets added only after the wrappers are updated with `register_extendr()`, so you could test that it's absent after `use_extendr()` and present after `register_extendr()`.

Having said that, I'm a bit concerned about these massive github action scrips we're writing. Maybe move this test into a separate script, just so things get a little more manageable?

**Review by clauswilke — APPROVED** [review]
Thanks!

**Comment by Ilia-Kosenkov** [comment]
@clauswilke,
Great, thank you, merging now.

**Pull request merged**